src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,104 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Loads and saves duke.txt.
+ */
+public class Storage {
+
+    /**
+     * Writes list of tasks into a text file.
+     *
+     * @param taskArrayList list of tasks.
+     */
+    public static void save(ArrayList<Task> taskArrayList) {
+        try {
+            File file = new File("data/duke.txt");
+            File directory = new File("data");
+
+            if (!directory.exists()) {
+                directory.mkdir();
+            }
+
+            FileWriter myWriter = new FileWriter(file);
+            for (Task t: taskArrayList) {
+                myWriter.write(t.toStringFileFormat() + System.lineSeparator());
+            }
+            myWriter.close();
+        } catch (IOException e) {
+            throw new DukeException("Save issue.");
+        }
+    }
+
+    /**
+     * Loads text file from memory.
+     *
+     * @return arraylist of type task.
+     * @throws IOException if file reading issues arise.
+     */
+    public static ArrayList<Task> loadTask() throws IOException {
+        ArrayList<Task> taskArrayList = new ArrayList<>();
+        try {
+            File taskFile = new File("data/duke.txt");
+            Scanner sc = new Scanner(taskFile);
+
+            while (sc.hasNext()) {
+                String taskString = sc.nextLine();
+                String[] taskStringInArray = taskString.split(" \\| ");
+                String taskType = taskStringInArray[0];
+                if (taskStringInArray.length > 1) {
```

Comment:
Perhaps you could consider the "Make the app path prominent" technique here by changing the inequality. This might also helps to make the conditions less nested.